streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1499,6 +1503,93 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
         assertFalse(testStreamTask.committed);
     }
 
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldAlwaysUpdateWithLatestTopicsFromStreamPartitionAssignor() throws Exception {
+        final TopologyBuilder topologyBuilder = new TopologyBuilder();
+        topologyBuilder.addSource("source", Pattern.compile("t.*"));
+        topologyBuilder.addProcessor("processor", new MockProcessorSupplier(), "source");
+
+        final StreamThread thread = new StreamThread(
+            topologyBuilder,
+            config,
+            clientSupplier,
+            applicationId,
+            clientId,
+            processId,
+            metrics,
+            mockTime,
+            new StreamsMetadataState(topologyBuilder, StreamsMetadataState.UNKNOWN_HOST),
+            0);
+
+        final StreamPartitionAssignor partitionAssignor = new StreamPartitionAssignor();
+        final Map<String, Object> configurationMap = new HashMap<>();
+
+        configurationMap.put(StreamsConfig.InternalConfig.STREAM_THREAD_INSTANCE, thread);
+        configurationMap.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 0);
+        partitionAssignor.configure(configurationMap);
+
+        thread.setPartitionAssignor(partitionAssignor);
+
+        final Field
+            nodeToSourceTopicsField =
+            topologyBuilder.getClass().getDeclaredField("nodeToSourceTopics");
+        nodeToSourceTopicsField.setAccessible(true);
```

Comment:
Arg. Reflections. Seems like our code is not well structure if we need this. Can you file a Jira to do some follow up cleanup (or if this does not make it into 0.11, we might to the cleanup first).